pkg/controller/datadogagentdeployment/agent.go

```diff
@@ -130,7 +134,7 @@ func (r *ReconcileDatadogAgentDeployment) createNewExtendedDaemonSet(logger logr
 	// ExtendedDaemonSet up to date didn't exist yet, create a new one
 	var newEDS *edsdatadoghqv1alpha1.ExtendedDaemonSet
 	var hash string
-	if newEDS, hash, err = newExtendedDaemonSetFromInstance(logger, agentdeployment); err != nil {
+	if newEDS, hash, err = newExtendedDaemonSetFromInstance(logger, agentdeployment, nil); err != nil {
```

Comment:
I think we don't have the same limitation with ExtendedDaemonset, but it better to use the same logic 👍